data/src/message.rs

```diff
@@ -52,6 +52,18 @@ impl Message {
         let text = text(&proto)?;
         let prefix = proto.prefix?;
 
+        // Check to see if we have timestamp on a tag.
+        // TODO: not sure why it isn't always working.
+        let timestamp = proto
+            .tags
+            .as_ref()
+            .and_then(|tags| tags.iter().find(|tag| tag.0 == "time"))
+            .and_then(|tag| tag.1.clone())
+            .and_then(time::Posix::parse_rfc3339)
+            .unwrap_or_else(time::Posix::now);
```

Comment:
i would assume we get timestamp on all messages after enabling the capability but that seems not to be the case.